moengage-segment-kotlin-destination/src/main/java/com/segment/analytics/kotlin/destinations/moengage/MoEngageDestination.kt

```diff
@@ -0,0 +1,250 @@
+import org.json.JSONObject
+
+private const val MOENGAGE_SEGMENT_INTEGRATION_KEY = "MoEngage"
+private const val INTEGRATION_META_TYPE = "segment"
+private const val USER_TRAIT_ANONYMOUS_ID = "anonymousId"
+private const val USER_TRAIT_USER_EMAIL = "email"
+private const val USER_TRAIT_UNIQUE_ID = "userId"
+private const val USER_TRAIT_NAME = "name"
+private const val USER_TRAIT_USER_MOBILE = "phone"
+private const val USER_TRAIT_FIRST_NAME = "firstName"
+private const val USER_TRAIT_LAST_NAME = "lastName"
+private const val USER_TRAIT_USER_GENDER = "gender"
+private const val USER_TRAIT_USER_BDAY = "birthday"
+private const val USER_TRAIT_USER_ADDRESS = "address"
+private const val USER_TRAIT_USER_ADDRESS_CITY = "city"
+private const val USER_TRAIT_USER_ADDRESS_STATE = "state"
+private const val USER_TRAIT_USER_ADDRESS_COUNTRY = "country"
+private const val USER_TRAIT_USER_LOCATION = "location"
+private const val USER_TRAIT_USER_LOCATION_LONGITUDE = "longitude"
+private const val USER_TRAIT_USER_LOCATION_LATITUDE = "latitude"
+private const val USER_ATTRIBUTE_SEGMENT_ID = "USER_ATTRIBUTE_SEGMENT_ID"
+
+@kotlinx.serialization.Serializable
+data class MoEngageSettings(val apiKey: String)
+
+class MoEngageDestination(private val application: Application) : DestinationPlugin(),
+    VersionedPlugin {
+
+    private lateinit var integrationHelper: MoEIntegrationHelper
+    private lateinit var instanceId: String
+
+
+    companion object {
+        private const val tag = "MoEngageDestination_" + BuildConfig.MOENGAGE_SEGMENT_KOTLIN_VERSION
+
+        val mapper: Map<String, String> = mapOf(
+            USER_TRAIT_ANONYMOUS_ID to USER_ATTRIBUTE_SEGMENT_ID,
+            USER_TRAIT_USER_EMAIL to USER_ATTRIBUTE_USER_EMAIL,
+            USER_TRAIT_UNIQUE_ID to USER_ATTRIBUTE_UNIQUE_ID,
+            USER_TRAIT_NAME to USER_ATTRIBUTE_USER_NAME,
+            USER_TRAIT_USER_MOBILE to USER_ATTRIBUTE_USER_MOBILE,
+            USER_TRAIT_FIRST_NAME to USER_ATTRIBUTE_USER_FIRST_NAME,
+            USER_TRAIT_LAST_NAME to USER_ATTRIBUTE_USER_LAST_NAME,
+            USER_TRAIT_USER_GENDER to USER_ATTRIBUTE_USER_GENDER,
+            USER_TRAIT_USER_BDAY to USER_ATTRIBUTE_USER_BDAY
+
+        )
+    }
+
+    override val key: String
+        get() = MOENGAGE_SEGMENT_INTEGRATION_KEY
+
+    override fun update(settings: Settings, type: Plugin.UpdateType) {
+        try {
+            super.update(settings, type)
+            Logger.print { "$tag update(): in" }
+            if (type == Plugin.UpdateType.Initial) {
+                if (settings.hasIntegrationSettings(key)) {
+                    val moEngageSettings: MoEngageSettings =
+                        settings.destinationSettings(key) ?: return
+                    instanceId = moEngageSettings.apiKey
+                    integrationHelper =
+                        MoEIntegrationHelper(application, IntegrationPartner.SEGMENT)
+                    integrationHelper.initialize(instanceId, application)
```

Comment:
Don't use static import for functions of the SDKs

Reply:
Done